### ISSUE TITLE
async rule chain execution 

### ISSUE THREAD

**Issue by ariellev** [maticzav/graphql-shield#331]
# Bug report

* [ ] I have checked other issues to make sure this is not a duplicate.

## Describe the bug

Bug in current graphql-shield 5.3.2 prohibits an async rule chain execution.
Symptom showing an execution of a subsequential rule before a completion of a preceding rule

3. I use these permissions

```ts
const isAuthenticatedUser = rule({ cache: 'contextual' })(
  async (parent: any, args: any, context: any) => {
..
const data: any = await prisma.user({ context.id })

// the execution breaks here and jumps to the next rule.
// nevertheless the block will be run to the end eventually
..
}

const permissions = shield({
  Query: {
    book: ...,
  },
  Book: {
    content: and(isAuthenticatedUser, rules.isBusiness, rules.isAdmin),
  },
})
```

## Expected behavior

Async rules run until completion before control passes over to the following rule

**Comment by maticzav** [comment]
Hey 👋,

Thanks for opening an issue. If I am not mistaken, the reason you are experiencing it is due to parallel execution of rules in the newer versions of `graphql-shield`.

Could you verify that your issue is not connected to #267? Otherwise, we can continue the discussion in that thread.

**Comment by maticzav** [comment]
Fixed by #392
